core/src/main/scala/org/apache/spark/ui/scope/RDDOperationGraph.scala

```diff
@@ -72,6 +72,22 @@ private[ui] class RDDOperationCluster(val id: String, private var _name: String)
   def getCachedNodes: Seq[RDDOperationNode] = {
     _childNodes.filter(_.cached) ++ _childClusters.flatMap(_.getCachedNodes)
   }
+
+  def canEqual(other: Any): Boolean = other.isInstanceOf[RDDOperationCluster]
+
+  override def equals(other: Any): Boolean = other match {
+    case that: RDDOperationCluster =>
+      (that canEqual this) &&
+          _childClusters == that._childClusters &&
+          id == that.id &&
+          _name == that._name
+    case _ => false
+  }
+
+  override def hashCode(): Int = {
+    val state = Seq(_childClusters, id, _name)
+    state.map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
```

Comment:
I updated this to use `java.util.Objects.hashCode`, assuming you meant that and not the Guava Objects.hashCode. I opted for this form because it was fairly concise and there doesn't seem to be an established common practice in the rest of the codebase from a quick search. Just let me know what the consensus on this is and I'll change it. Thanks!